### PR TITLE
#152 Use higher quality images for country flags

### PR DESCRIPTION
**Pull request by sol695510**
### Description of the Change

All 218 currently supported country flags used as icons in Constellation have been updated with higher resolution versions.

Additionally a "flags" folder was added to the "icons" folder in the directory path "constellation\CoreVisual\release\modules\ext\icons" for ease of alteration and testing. FlagIconProvider.java was modified accordingly to reflect the directory location change of the flag images.

### Alternate Designs

An alternate set was initially proposed as it had all 218 flags in a consistent style, from here [https://github.com/gosquared/flags](url). While the flags were more accurate with better detail, the maximum resolution of the set was 64x64. It was advised that at that resolution the images would be upscaled and the improvement in quality would become negligible.

The current sets chosen contain flags which are 512x512 or greater which have been resized to 256x256, the maximum resolution currently supported by Constellation.

216/218 were sourced from here [https://www.flaticon.com/packs/international-flags/2](url).

Guyana's flag was sourced from here [https://www.flaticon.com/packs/flags-15](url).

A matching style and high resolution version for the New Caledonia flag was not found from the same site the other sets were found, so a 1280x640 image from [https://en.wikipedia.org/wiki/Kanak_people](url) was used.

### Why Should This Be In Core?

More accurate depiction of country flags.

### Benefits

Flags are more accurate and previously some flags had been replicated to compensate for missing flags, such as the reuse of the Liberia flag for the United States of America.

The flags are also more consistent now with the slight exception of the New Caledonia flag.

### Possible Drawbacks

Higher resolution may now increase startup and load times of Constellation and graphs.

### Verification Process

Functionality was verified by opening a new graph in Constellation and navigating to "Experimental" > "Build Graph" > "Sphere Graph Builder" and visually checking the whether the code and file changes were taking effect.

### Applicable Issues

#152

### CODE REVIEW

**Comment by arcturus2** [comment]
Very nice @sol695510. Thanks for the hard work to find all of these icons!

Confirming that the US flag has the stars showing up now which was what triggered this PR:

image

FYI you can do Experimental -> Build Graph -> Icon Graph to see all of the icons too.

File: CoreVisual/src/au/gov/asd/tac/constellation/visual/icons/FlagIconProvider.java [hunk 1]
```diff
@@ -28,883 +28,883 @@
 @ServiceProvider(service = ConstellationIconProvider.class)
 public class FlagIconProvider implements ConstellationIconProvider {
 
-    public static final ConstellationIcon AFGHANISTAN = new ConstellationIcon.Builder("Afghanistan", new FileIconData("modules/ext/icons/afghanistan.png", "au.gov.asd.tac.constellation.visual"))
+    public static final ConstellationIcon AFGHANISTAN = new ConstellationIcon.Builder("Afghanistan", new FileIconData("modules/ext/icons/flags/afghanistan.png", "au.gov.asd.tac.constellation.visual"))
```
Comment (arcturus2) [inline CoreVisual/src/au/gov/asd/tac/constellation/visual/icons/FlagIconProvider.java]: Thanks @sol695510, creating the flags folder was something I wanted to see done too. Nice 👍

**Comment by sol695510** [comment]
Hi arcturus2,

in regards to the file attribution.md, I've added 3 new attributions but I
wanted to clarify whether these two attributions which I've highlighted in
the screenshot should be removed.

[image: attribution inquiry.PNG]

The URL goes to the flag icon website but I don't recognise the 2 .zips
mentioned, flat-flat-2.zip and map_markers.zip. I thought they were the old
flags but they don't seem to be so I'm not sure whether to remove them.

Sincere regards,
sol695510

**Comment by arcturus2** [comment]
> Hi arcturus2, in regards to the file attribution.md, I've added 3 new attributions but I wanted to clarify whether these two attributions which I've highlighted in the screenshot should be removed. [image: attribution inquiry.PNG] The URL goes to the flag icon website but I don't recognise the 2 .zips mentioned, flat-flat-2.zip and map_markers.zip. I thought they were the old flags but they don't seem to be so I'm not sure whether to remove them. Sincere regards, sol695510

Unfortunately I can't see the screenshot but you can get rid of the references to `flat-flat-2.zip` and `map_markers.zip`.

**Comment by sol695510** [comment]
attribution inquiry

@arcturus2 here's the picture, not sure why it didn't upload. I'll remove them the unneeded references.

**Pull request merged**